core/modules/taxonomy/taxonomy.module

```diff
@@ -1482,11 +1489,124 @@ function taxonomy_field_formatter_view($entity_type, $entity, $field, $instance,
         );
       }
       break;
+
+    case 'taxonomy_term_reference_rendered':
+      // To prevent infinite recursion caused by reference cycles, we store
+      // diplayed terms in a recursion queue.
+      $recursion_queue = &backdrop_static(__FUNCTION__, array());
+
+      // If no 'referencing entity' is set, we are starting a new 'reference
+      // thread' and need to reset the queue.
+      // @todo Bug: $entity->referencing_entity on terms referenced in a different
+      // thread on the page. E.g: 1 references 1+2 / 2 references 1+2 / visit homepage.
+      // We'd need a more accurate way...
+      if (!isset($entity->referencing_entity)) {
+        $recursion_queue = array();
+      }
+
+      // The recursion queue only needs to track terms.
+      if (in_array($entity_type, array('node', 'user', 'taxonomy_term'))) {
```

Comment:
Yeah that was language from node_reference. Since we can add term references to nodes, users and terms I had added that to the array. Maybe just remove the condition entirely.